src/components/select-utils.js

```diff
@@ -63,23 +63,23 @@ export const joinPostData = state => postId => {
 
   // Get the list of post's recipients
   const recipients = post.postedTo
-    .map((subscriptionId) => {
-      let userId = (state.subscriptions[subscriptionId]||{}).user
-      let subscriptionType = (state.subscriptions[subscriptionId]||{}).name
-      if (userId === post.createdBy && subscriptionType === 'Directs') {
-        // Remove "directs to yourself" from the list
-        return false
-      } else {
-        return userId
-      }
+    .map(subscriptionId => {
+      const userId = (state.subscriptions[subscriptionId]||{}).user
+      const subscriptionType = (state.subscriptions[subscriptionId]||{}).name
+      return userId !== post.createdBy || subscriptionType !== 'Directs' ? userId : false
```

Comment:
Actually, I think in this case it's better to be more verbose for the sake of comprehension and leave the condition not inverted, i.e.
```
(userId === post.createdBy && subscriptionType === 'Directs')
```

There's still a room for improvement, of course, like removing that else-return. So my final suggested change would be:
```
    .map((subscriptionId) => {
      const userId = (state.subscriptions[subscriptionId]||{}).user
      const subscriptionType = (state.subscriptions[subscriptionId]||{}).name

      // Remove "directs to yourself" from the list
      if (userId === post.createdBy && subscriptionType === 'Directs') {
        return false
      }

      return userId
    }
```

I know it's 10 lines instead of 3, but I think verbosity pays off in this case.